cmd/salt-updater/main.go

```diff
@@ -89,11 +83,28 @@ func runMain() error {
 	// Don't want to run any salt commands before the device is registered as it will set a salt minion_id
 	if _, err := os.Stat("/etc/salt/minion_id"); os.IsNotExist(err) {
 		log.Println("The salt minion_id file was not found, meaning that the device has not registered yet, exiting.")
-		return nil
+		// return nil
+	}
+	saltState, _ := saltrequester.ReadStateFile()
+	nodegroupOut, _ := ioutil.ReadFile("/etc/cacophony/salt-nodegroup")
+	nodegroup := strings.TrimSpace(string(nodegroupOut))
+	if strings.TrimSpace(saltState.LastCallNodegroup) != nodegroup {
```

Comment:
Because the last call is what saves the nodegroup won't `saltState.LastCallNodegroup` and `nodegroup` always match?

Reply:
This just catches if you change the nodegroup on the salt master and run a state apply manually

Reply:
Yeah that makes sense